protocol/http/types.go

```diff
@@ -172,22 +154,10 @@ func (h ResponseHeader) Get(key string) (string, bool) {
 }
 
 // Set key-value pair in header map, the previous pair will be replaced if exists
-//
-// Due to the fact that fasthttp's implementation doesn't have correct semantic for
-//Set("key", "") and Peek("key") at the
-// first time of usage. We need another way for compensate.
-//
-// The problem is caused by the func initHeaderKV, if the original kv.value is nil, ant input value is also nil,
-// then the final kv.value remains nil.
-//
-// kv.value = append(kv.value[:0], value...)
-//
-// fasthttp do has the kv entry, but kv.value is nil, so Peek("key") return nil. But we want "" instead.
 func (h ResponseHeader) Set(key string, value string) {
-	if value == "" {
-		// Set a placeholder first, so that ResponseHeader can get this value after setting an empty value.
-		h.ResponseHeader.Set(key, PlaceHolder)
-	}
+	// pre-emptively delete because fasthttp doesn't handle overwriting
+	// set-cookie, yet
+	h.ResponseHeader.Del(key)
```

Comment:
we notice that this can lead to inconsistent behavior, either compared with the previous version or with the request header. 

And we should be careful with all possible keys, not only for `set-cookie`, plz ref to `RequestHeader.setSpecialHeader` and `ResponseHeader.setSpecialHeader` for more details.

It's  better to add detailed test cases (should contains all special keys) to the current commit first.

Reply:
ok I am in the middle of a release so comprehensively back-filling missing tests won't happen until after 31 oct.  I'll continue next week.